runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -463,6 +463,11 @@ public static void main(String[] args) throws Exception {
     // metrics.
     MetricsEnvironment.setProcessWideContainer(new MetricsLogger(null));
 
+    // When enabled, the Pipeline will record Per-Worker metrics that will be piped to WMW.
+    StreamingStepMetricsContainer.setEnablePerWorkerMetrics(
+        options.isEnableStreamingEngine()
+            && DataflowRunner.hasExperiment(options, "enable_per_worker_metrics"));
```

Comment:
Should this be an experiment, or just on by default?

Reply:
Will leave this as an experiment until I've done some load tests. Then I will enable it as default for Streaming Engine Jobs.